### DISCUSSION TITLE
A "Did you mean..." plugin for terminal users

### DISCUSSION THREAD

**Discussion by Penaz91** [nvim-mini/mini.nvim#42] [Ideas]
Greetings,

I'm opening a discussion because I don't really know if issues are the right place for this: I've been dabbling a bit with (n)vim plugins and I am an avid terminal user.

Many times I happen to be "in the zone" and type partial filenames and end up with an empty buffer instead of the filename I wanted to open (even though I swear I pressed `tab`).

So I've been dabbling with the idea of a plugin that, if you are creating a new file and there are files with a similar name (like "ini" for "init.vim"), they are shown via a `vim.ui.select` menu. If you press escape you just ignore the prompt and make a new file.

Could this interest some users? The proof of concept is currently undocumented but the whole code is under 100 lines, so I should not have any issues commenting it.

Thank you for your attention and for making these modules!

**Comment by echasnovski** [comment]
Hello!

I am not sure I completely understand this. Do you mean like the following:
- Inside command line type `nivim ini`.
- Instead of opening Neovim with single buffer for 'ini' file, look inside current directory (maybe recursive) for files with name "similar" to 'ini'. Somehow suggest them to user.

Something like that?

**Reply by Penaz91** [reply]
Yes. In my experiment I used vim glob, split and expand functions (taken from an old plugin that has been lost to memory) and "modernized it".

Here's a small example from my experimenting (sorry for the screen crunch!)

https://user-images.githubusercontent.com/6682630/151051331-aaca9ae3-583f-4c2b-9ab8-6e72acc9a7ff.mp4

**Reply by echasnovski** [reply]
I can't watch the video. It says that file is damaged.

**Reply by Penaz91** [reply]
It seems that there is an FFMPEG update rolling out, so some videos may be seen as "corrupted". I've had that issue with ArchLinux and Firefox (I just updated QuteBrowser, my main driver, and it works)

**Comment by echasnovski** [comment]
I just looked at your 'MiniDYM' plugin. The workflow seem to be to check for every new buffer if it is present or not, and suggest "Did you mean ..." in latter case.

Personally I (and probably quite decent percentage of people) quite rarely open files with `edit` from command line. Usually it is from `telescope.nvim`, sometimes - from `nvim-tree`.

If it only to be checked on Neovim start, it's better to look at `VimEnter` event instead of `BufNewFile`.

**Reply by Penaz91** [reply]
I usually either open files from terminal (so `nvim <filename>`) or using `netrw` or similar methods too, so the VimEnter check would make more sense.
